pkg/local/diff.go

```diff
@@ -190,8 +191,9 @@ func Undo(c Changes) Changes {
 // This walker function simulates the way object listing is performed by S3. Contrary to how a standard FS walk function behaves, S3
 // does not take into consideration the directory hierarchy. Instead, object paths include the entire path relative to the root and as a result
 // the directory or "path separator" is also taken into account when providing the listing in a lexicographical order.
-func WalkS3(root string, includeFolders bool, callbackFunc func(p string, info fs.FileInfo, err error) error) error {
+func WalkS3(root string, callbackFunc func(p string, info fs.FileInfo, err error) error) error {
 	var stringHeap StringHeap
+	var dirsInfo = make(map[string]os.FileInfo)
```

Comment:
Consider changing the StringHeap to a "FileInfo" heap, that way you can reduce the number of variables and avoid the possible nil pointer exception mentioned below.
It doesn't have to be on this PR but if not - let's open an improvement for it

Reply:
There's a single place where we push to the heap:
> heap.Push(&stringHeap, dir)

And right before that, I'm saving the dir info:
> dirsInfo[dir] = info

So in what scenario I'll get null for `dirsInfo[dir]`?
Am I missing something?

Reply:
In the scenario where the code will move a little bit and somehow the key will not exist in the map :)
Since the two data structures are not coupled in any way - we should add guardrails even if you don't see any way this will happen in the current state of the code

Reply:
I think that changing `StringHeap` into `FileInfoHeap` will be weird and even more confusing (and less reusable for sure).

Added null checks.